taichi/runtime/llvm/runtime.cpp

```diff
@@ -1046,7 +1047,7 @@ struct range_task_helper_context {
 
 void parallel_range_for_task(void *range_context, int task_id) {
   auto ctx = *(range_task_helper_context *)range_context;
-  char tls_buffer[taichi_tls_buffer_size];
+  alignas(8) char tls_buffer[ctx.tls_size];
```

Comment:
Wow I didn't even know we can do variable length array nowadays. Do you know if this is supported on C++17 now? (I did a quick search and found that while this is supported in C99, the support of VLA in C++ remains vague.. I tested with `clang` and it's working)

Reply:
I'm not sure if that's in C++17 TBH. However, LLVM `alloca` does support allocating arrays with the number of elements being a variable (instruction) :-)